components/camel-pdf/src/main/java/org/apache/camel/component/pdf/PdfProducer.java

```diff
@@ -78,6 +83,22 @@ public void process(Exchange exchange) throws Exception {
         exchange.getMessage().setBody(result);
     }
 
+    private OutputStream doMerge(Exchange exchange) throws IOException {
+        LOG.debug("Got {} operation, going to merge multiple files into a single pdf document.",
+                pdfConfiguration.getOperation());
+        PDFMergerUtility mergerUtility = new PDFMergerUtility();
+        List<File> files = exchange.getIn().getHeader(FILES_TO_MERGE_HEADER_NAME, new ArrayList<File>(), List.class);
```

Comment:
I think you can use `ExchangeHelper.getMandatoryHeader` here, so it's consistent with the rest of the code. It would also avoid creating the ArrayList, which should help avoid unnecessary pressure on the GC.